python_modules/libraries/dagster-k8s/dagster_k8s/pipes.py

```diff
@@ -101,13 +102,27 @@ class _PipesK8sClient(PipesClient):
             context into the k8s container process. Defaults to :py:class:`PipesEnvContextInjector`.
         message_reader (Optional[PipesMessageReader]): A message reader to use to read messages
             from the k8s container process. Defaults to :py:class:`PipesK8sPodLogsMessageReader`.
+        load_incluster_config (Optional[bool]): Whether this client is expected to be running from inside
+            a kubernetes cluster and should load config using ``kubernetes.config.load_incluster_config``.
+            Otherwise ``kubernetes.config.load_kube_config`` is used with the kubeconfig_file argument.
+            Default: None
+        kubeconfig_file (Optional[str]): The value to pass as the config_file argument to
+            ``kubernetes.config.load_kube_config``.
+            Default: None.
+        kube_context (Optional[str]): The value to pass as the context argument to
+            ``kubernetes.config.load_kube_config``.
```

Comment:
To joe's point are we providing concrete benefit here over a user calling one of these methods themselves in the callsite? load_kube_config has other arguments too https://github.com/kubernetes-client/python/blob/c6acb1f04a40ae4a44f939b5b1885f9809247c31/kubernetes/base/config/kube_config.py#L795

One observation is that the default behavior below if none of these are set, while cool, makes it impossible to customize your own config loading behavior outside of the function if you want

Reply:
> a user calling one of these methods themselves in the callsite? 

I think the two things are:
* easy migration from our APIs or airflow APIs which offer similar controls 
* having this be part of the resource so it gets handled as part of the resource instead of in each op/asset that its used 


> makes it impossible to customize your own config loading behavior outside of the function if you want

this is a good point